test/integration/create-loaded.int.c

```diff
@@ -0,0 +1,118 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <inttypes.h>
+
+#define LOGMODULE test
+#include "log/log.h"
+#include "sapi-util.h"
+
+TSS2_RC
+test_invoke (TSS2_SYS_CONTEXT *sapi_context)
+{
+    TSS2_RC                 rc              = TSS2_RC_SUCCESS;
+    TPM2B_SENSITIVE_CREATE  in_sensitive    = { 0 };
+    TPM2B_PUBLIC            in_public       = { 0 };
+    TPM2B_PRIVATE           out_private     = { 0 };
+    TPM2B_PUBLIC            out_public      = { 0 };
+    TPM2B_NAME              name            = TPM2B_NAME_INIT;
+    TPM2B_NAME              qualified_name  = TPM2B_NAME_INIT;
+    TPM2_HANDLE             object_handle   = 0;
+    TSS2L_SYS_AUTH_COMMAND  auth_cmd = {
+        .auths = {{ .sessionHandle = TPM2_RS_PW }},
+        .count = 1
+    };
+    TSS2L_SYS_AUTH_RESPONSE auth_rsp = {
+        .count = 0
+    };
+
+    if (sapi_context == NULL)
+        return TSS2_RC_LAYER_MASK | TSS2_BASE_RC_BAD_REFERENCE;
+
+    in_public.publicArea.type = TPM2_ALG_RSA;
+    in_public.publicArea.nameAlg = TPM2_ALG_SHA256;
+    in_public.publicArea.objectAttributes |= TPMA_OBJECT_RESTRICTED;
+    in_public.publicArea.objectAttributes |= TPMA_OBJECT_USERWITHAUTH;
+    in_public.publicArea.objectAttributes |= TPMA_OBJECT_DECRYPT;
+    in_public.publicArea.objectAttributes |= TPMA_OBJECT_FIXEDTPM;
+    in_public.publicArea.objectAttributes |= TPMA_OBJECT_FIXEDPARENT;
+    in_public.publicArea.objectAttributes |= TPMA_OBJECT_SENSITIVEDATAORIGIN;
+    in_public.publicArea.parameters.rsaDetail.symmetric.algorithm = TPM2_ALG_AES;
+    in_public.publicArea.parameters.rsaDetail.symmetric.keyBits.aes = 128;
+    in_public.publicArea.parameters.rsaDetail.symmetric.mode.aes = TPM2_ALG_CFB;
+    in_public.publicArea.parameters.rsaDetail.scheme.scheme = TPM2_ALG_NULL;
+    in_public.publicArea.parameters.rsaDetail.keyBits = 2048;
+
+    /* Create an object using CreateLoaded.
+     * The result should be that the created object
+     * stays in the TPM
+     */
+    LOG_INFO("Calling CreateLoaded");
+    rc = Tss2_Sys_CreateLoaded (sapi_context,
+                                TPM2_RH_OWNER,
+                                &auth_cmd,
+                                &in_sensitive,
+                                &in_public,
+                                &object_handle,
+                                &out_private,
+                                &out_public,
+                                &name,
+                                &auth_rsp);
+    if (rc == TPM2_RC_SUCCESS) {
+        LOG_INFO("success object handle: 0x%x", object_handle);
+    } else {
+        LOG_ERROR("CreateLoaded FAILED! Response Code : 0x%x", rc);
+        exit(1);
+    }
+
+    memset(&out_public, '\0', sizeof(out_public));
+    memset(&name, '\0', sizeof(name));
```

Comment:
can you, or don't you already do this when the structure is initialized?

Reply:
I agree that we can have some common init helpers, but in this case it is not redundant.
Both of them will be populated with data by the Tss2_Sys_CreateLoaded(). I want to clean them fresh before calling to Tss2_Sys_ReadPublic()